integration-tests/typescript-angular/src/app/todo-lists.yaml/client.service.ts

```diff
@@ -36,19 +48,21 @@ export class ApiClient {
     )
   }
 
-  private _queryParams(
-    queryParams: Record<
-      string,
-      boolean | number | string | string[] | undefined | null
-    >
-  ): HttpParams {
-    const result = new HttpParams()
-    Object.entries(queryParams).forEach(([name, value]) => {
-      if (value !== undefined && value !== null) {
-        result.set(name, String(value))
+  private _queryParams(queryParams: QueryParams): HttpParams {
+    return Object.entries(queryParams).reduce((result, [name, value]) => {
```

Comment:
Turns out that the `HttpParams` object is immutable, and so this was previously totally broken.

ref: https://angular.io/api/common/http/HttpParams#description
> This class is immutable; all mutation operations return a new instance.